ui/org.eclipse.pde.bnd.ui/src/org/eclipse/pde/bnd/ui/RepositoryUtils.java

```diff
@@ -70,6 +75,14 @@ public static List<RepositoryPlugin> listRepositories(final Workspace localWorks
 //				Workspace bndWorkspace = Central.getWorkspaceIfPresent();
 //				if ((bndWorkspace == localWorkspace) && !bndWorkspace.isDefaultWorkspace())
 //					repos.add(Central.getWorkspaceRepository());
+				
+				// TODO this is not perfect, because it is only working
+				// if you are selecting a bnd project. Would be better if bnd WorkspaceRepository is added always
+				// e.g. if there is at least one bnd project
+				if (WorkspaceLayout.BND == localWorkspace.getLayout() && !localWorkspace.isDefaultWorkspace()) {
+					repos.add(localWorkspace.getWorkspaceRepository());
```

Comment:
PDE also has a "Workspace repository" so maybe we should extend the Workspace class and use it in PDE to return the real Workspace repo there?

Reply:
This is something what I would like to talk about with you. I have some difficulty understanding the current "thinking" behind all this workspaces (Eclipse Workspace with different projects, partly projects imported from a bnd workspace... a bit confusing :) )

Reply:
Btw: this change here makes the Bnd Workspace Repo show up:

<img width="359" alt="image" src="https://github.com/user-attachments/assets/0aaff715-4c5a-45d0-a1b3-3fd60026c25f" />

Reply:
> Btw: this change here makes the Bnd Workspace Repo show up

:+1:  it seems the icon for this is currently missing.

Reply:
hmm now that you mention it :) I can have a look.

Reply:
I added the icon. I duplicated "repository.png" into "workspacerepo.png". They now use the same icon.